src/nvim/ex_cmds.c

```diff
@@ -224,15 +225,17 @@ void ex_align(exarg_T *eap)
          */
         if (has_tab)
           while (new_indent > 0) {
-            (void)set_indent(new_indent, 0);
+            assert(new_indent <= INT_MAX);
+            (void)set_indent((int)new_indent, 0);
             if (linelen(NULL) <= width) {
               /*
                * Now try to move the line as much as possible to
                * the right.  Stop when it moves too far.
                */
-              do
-                (void)set_indent(++new_indent, 0);
-              while (linelen(NULL) <= width);
+              do {
+                assert(new_indent + 1 <= INT_MAX);
+                (void)set_indent((int)++new_indent, 0);
```

Comment:
`++` and `--` need to go after the name if the distinction is not meaningful (from style guide). Although if `new_indent` is reverted this can be reverted entirely.